lib/utils/addon-utils.ts

```diff
@@ -132,7 +132,7 @@ export function conflictsWithAutoMode(minExpectedVersion: string | null) {
       if (minExpectedVersion === "fail") {
         throw new Error(`Deploying ${stack} failed. This add-on is already available on the cluster with EKS Auto Mode.`);
       }
-      else if (minExpectedVersion == null) {
+      else if (minExpectedVersion == null || this.version == null || this.version == "auto") {
```

Comment:
this fixes the runtime but incorrect functionally. 
CoreDNS has the following decorator:  @utils.conflictsWithAutoMode("v1.9.3-eksbuild.7")
The expectation is that  it should compare the versions. 

So for all the derivatives of CoreAddOn - I suppose we can also add version as the member variable.

I propose to keep this fix and apply the change to the core addons so that we can produce meaningful messages when needed. E.g. if they deploy coredns explicitly with the version that does not conflict, we should not warn.